src/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,157 @@
+"use client";
+
+import React from "react";
+import {
+  useSidebarStateContext,
+  useSidebarActionsContext,
+} from "@context/SidebarContext";
+import {
+  Drawer,
+  Box,
+  Typography,
+  List,
+  ListItemButton,
+  Divider,
+} from "@mui/material";
+import theme from "theme";
+import { getCourses } from "@services/client/course";
+import { useUserSession } from "@context/UserSessionContext";
+import { useRouter } from "next/navigation";
+import { usePathname } from "next/navigation";
+import { IdentifiableCourse } from "@interfaces/type";
+import MenuButton from "./buttons/MenuButton";
+import ModeEditOutlinedIcon from "@mui/icons-material/ModeEditOutlined";
+
+const Sidebar = React.memo(() => {
+  const [courses, setCourses] = React.useState<IdentifiableCourse[]>([]);
+  const { isOpen } = useSidebarStateContext();
+  const { closeSidebar } = useSidebarActionsContext();
+  const { user } = useUserSession();
+  const { courses: courseIds = [] } = user || {};
+  const router = useRouter();
+  const pathname = usePathname();
+
+  React.useEffect(() => {
+    const loadCourses = async () => {
+      const allCourses = await getCourses();
+      const filteredCourses = allCourses
+        .filter((course) => courseIds.includes(course.id))
+        .map((course) => course);
+      setCourses(filteredCourses);
+    };
+    loadCourses();
+  }, [user]);
+
+  const handleCourseClick = (course: IdentifiableCourse) => {
+    const isCourseRoute = /\/[a-zA-Z]+\d+/.test(pathname);
+
+    const basePath = isCourseRoute
+      ? pathname.replace(/\/[a-zA-Z]+\d+.*$/, "")
+      : pathname;
+
+    router.push(`${basePath}/${course.id}`);
+    closeSidebar();
+  };
+
+  const handleManageCoursesClick = () => {
+    const isCourseRoute = /\/[a-zA-Z]+\d+/.test(pathname);
+
+    const basePath = isCourseRoute
+      ? pathname.match(/(.*\/[a-zA-Z]+\d+)/)?.[0]
+      : `${pathname}/${courses[0].id}`;
+
+    router.push(`${basePath}/profile/edit`);
```

Comment:
Can you clarify why we need to dynamically figure out the base path?

Reply:
/profile lives inside /[courseId] and sidebar is also accessible in /course so if in /course, it needs to go to /[courseId]/profile/edit so i'm thinking we take the first of courseIds from userSession as part of the basePath. lmk what you think

Reply:
Can't we always do 

```ts
route.push(`/private/${currentCourseId}/profile/edit
```
As a separate note, it feels weird that we have profile nested in `/[courseId]`.